drivers/scheduler/spec/factory.go

```diff
@@ -20,8 +20,10 @@ var appSpecFactory = make(map[string]*AppSpec)
 
 // register registers a new spec with the factory
 func (f *Factory) register(id string, app *AppSpec) {
-	logrus.Infof("Registering app: %v", id)
-	appSpecFactory[id] = app
+	if _, ok := appSpecFactory[id]; !ok {
+		logrus.Infof("Registering app: %v", id)
```

Comment:
Wanted this to stay because people will give their custom spec dirs and I want them to always see their applications getting identified on torpedo boots.